src/context/TopBanner.tsx

```diff
@@ -0,0 +1,73 @@
+import React, {PropsWithChildren, PropsWithoutRef} from 'react';
+import { useClientEffect } from "../utils/hookts"
+
+interface ITopBannerMessage {
+  text: string
+  from: number
+}
+
+interface ITopBannerProperties {
+  history: Array<ITopBannerMessage>
+}
+
+const defaultProperties: ITopBannerProperties = {
+   history: [],
+ };
+
+export interface ITopBannerContext extends ITopBannerProperties {
+  addMessage: (m: ITopBannerMessage) => void 
+  update: (key: keyof ISettingsProperties, value: any) => void
+}
+
+const defaultContext: ITopBannerContext = {
+ ...defaultProperties,
+  addMessage: () => {}, 
+  update: () => {},
+};
+
+export const TopBannerContext = React.createContext<ITopBannerContext>(defaultContext);
+
+export const TopBannerProvider = ({ children }):PropsWithChildren<{}> => {
+  const [context, setContext] = React.useState<ITopBannerContext>(defaultContext);
+  const ref = React.useRef<ITopBannerContext>(context);
+  
+  const updateContext = (ctx:ITopBannerContext) => {
+     setContext({
+       ...ctx,
+     });
+     ref.current = ctx;
+   };
+
+  const update = (key: keyof ITopBannerContext, value: any) => {
+     const newContext = {
+       ...ref.current,
+       [key]: value,
+     };
+     updateContext(newContext);
+     localStorage.setItem('top-banner-history', JSON.stringify(newContext));
```

Comment:
any strong opinions on naming the local storage entity?

Reply:
You can use the src/context/Theme.tsx (which is the global settings context) and add two variables
- hashMessage - save the content of the message as an hash (or not, it's just for easy comparaison later)
- hideTopBanner - save the status of the top banner

whenever the hashMessage is different from the currentMessageHash loaded from the .env variable, you reset hideTopBanner, and update the hashMessage variable

Reply:
Thanks for the input—Sure lets do this! What do you think about the value extraction of the message value? Currently we are providing span with `dangerouslySetInnerHTML` as the children to the `Warning`. Thats why I am extracing the `textContent` of the node via ref callback. But this now would make it inconsistent to compare the hashed `textContent` with the hashed .env variable. 

I think we should set the type of `Warning.children` to `string` and provide the env variable as string to the `Warning`.  Then we can make a simple comparison against the .env variables. Please let me know what you think